ui/src/criteria/attribute.tsx

```diff
@@ -0,0 +1,131 @@
+import { Checkbox, FormControlLabel, ListItem } from "@mui/material";
+import Stack from "@mui/material/Stack";
+import Typography from "@mui/material/Typography";
+import { CriteriaConfig, CriteriaPlugin, registerCriteriaPlugin } from "cohort";
+import produce from "immer";
+import React from "react";
+import * as tanagra from "tanagra-api";
+import { useUnderlay } from "../hooks";
+
+type Selection = {
+  id: number;
+  name: string;
+};
+
+interface Config extends CriteriaConfig {
+  entity: string;
+}
+
+interface Data extends Config {
+  selected: Selection[];
+}
+
+type AttributeEditProps = {
+  dispatchFn: (data: Data) => void;
+  data: Data;
+};
+
+@registerCriteriaPlugin("attribute", (config: CriteriaConfig) => ({
+  ...(config.plugin as Config),
+  selected: [],
+}))
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+class _ implements CriteriaPlugin<Data> {
+  public data: Data;
+
+  constructor(public id: string, data: unknown) {
+    this.data = data as Data;
+  }
+
+  renderEdit(dispatchFn: (data: Data) => void) {
+    return <AttributeEdit dispatchFn={dispatchFn} data={this.data} />;
+  }
+
+  renderDetails() {
+    return <ConceptDetails data={this.data} />;
+  }
+
+  generateFilter(entityVar: string, fromOccurrence: boolean) {
+    if (this.data.selected.length === 0) {
+      return null;
+    }
+    console.log(entityVar);
+    console.log(fromOccurrence);
+    return null;
+  }
+
+  occurrenceEntities() {
+    return [];
+  }
+}
+
+function AttributeEdit(props: AttributeEditProps) {
+  const underlay = useUnderlay();
+  const attributeName = props.data.entity;
+
+  const hintDisplayName = (hint: tanagra.EnumHintValue) =>
+    hint.displayName || "Unknown Value";
+
+  const enumHintValues = underlay.entities
+    .find((g) => g.name === underlay.primaryEntity)
+    ?.attributes?.find((attribute) => attribute.name === attributeName)
+    ?.attributeFilterHint?.enumHint?.enumHintValues;
+
+  const selectionIndex = (hint: tanagra.EnumHintValue) =>
+    props.data.selected.findIndex(
+      (row) => row.id === hint.attributeValue?.int64Val
+    );
+
+  return (
```

Comment:
Add a check for `enumHintValues?.length === 0` and display a message so that it's not a completely blank page in that case.

Reply:
correction --
add one more if sentence from line 78. if ===0, return <div>No attribute information</div> else return ...
Not sure whether needs warning or pop up windows.